src/persistence.ts

```diff
@@ -103,6 +113,36 @@ function ExtensionDataV2<ItemType extends z.ZodTypeAny>(itemSchema: ItemType) {
   });
 }
 
+export const WaitingSchemeV2 = z
+  .object({
+    waitTime: z.number().nonnegative().describe("wait time in minutes"),
+    weightMin: z
+      .number()
+      .nonnegative()
+      .describe("the weighted time for weighted random in minutes")
+      .optional(),
+    weightMsec: z
+      .number()
+      .gte(0)
```

Comment:
> in this case I like .gte(0) better because it makes the range very explicit.

Ah, good point.

> This makes me realize that these numbers aren't constraint of being whole integers, so maybe adding .int() would be a good idea.

Also a good point. I'm not sure how fractional milliseconds would work there, and I don't think we need to learn the answer :p